ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -28,76 +32,167 @@ const model: Model = {
+            data: {
+                id: '__MzQuMTIwLjAuMC8xNg',
+                label: 'Google/global | 34.120.0.0/16',
+                type: 'node',
+                width: 75,
+                height: 75,
+                entityType: 'CIDR_BLOCK',
+            },
+        },
+        {
+            id: 'afa12424-bde3-4313-b810-bb463cbe8f90',
+            label: 'External entities',
+            type: 'node',
+            width: 75,
+            height: 75,
+            data: {
+                id: 'afa12424-bde3-4313-b810-bb463cbe8f90',
+                label: 'External entities',
+                type: 'node',
+                width: 75,
+                height: 75,
+                entityType: 'EXTERNAL_ENTITIES',
             },
         },
         {
-            id: 'group1',
+            id: 'e8dabcb7-f471-414e-a999-fe91be5a28fa',
             type: 'group',
-            children: ['n1', 'n2'],
+            children: [
+                'e337f873-64d8-46be-84ed-2a0c38c75fac',
+                '09134b5d-8c12-41e8-821b-c97a5a1331c9',
+            ],
             group: true,
             label: 'stackrox',
             style: { padding: 15 },
             data: {
                 collapsible: true,
                 showContextMenu: false,
+                entityType: 'NAMESPACE',
+            },
+        },
+        {
+            id: 'EXTERNAL',
+            type: 'group',
+            children: ['__MzQuMTIwLjAuMC8xNg', 'afa12424-bde3-4313-b810-bb463cbe8f90'],
+            group: true,
+            label: 'External to cluster',
+            style: { padding: 15 },
+            data: {
+                collapsible: true,
+                showContextMenu: false,
+                entityType: 'EXTERNAL',
             },
         },
     ],
     edges: [
         {
             id: 'e1',
             type: 'edge',
-            source: 'n1',
-            target: 'n2',
+            source: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+            target: '09134b5d-8c12-41e8-821b-c97a5a1331c9',
         },
         {
             id: 'e2',
             type: 'edge',
-            source: 'n2',
-            target: 'n1',
+            source: '09134b5d-8c12-41e8-821b-c97a5a1331c9',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+        },
+        {
+            id: 'e3',
+            type: 'edge',
+            source: '__MzQuMTIwLjAuMC8xNg',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+        },
+        {
+            id: 'e4',
+            type: 'edge',
+            source: 'afa12424-bde3-4313-b810-bb463cbe8f90',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
         },
     ],
 };
 
+function findEntityById(graphModel, id) {
+    let entity = null;
+    entity = graphModel.nodes?.find((node) => node.id === id);
+    if (!entity) {
+        entity = graphModel.groups?.find((group) => group.id === id);
+    }
+    return entity;
+}
+
+function getUrlParamsForEntity(selectEntity): [string, string] {
```

Comment:
We should probably add a type for the `selectedEntity`. I think this might require us to make a custom type for our `Model`. The `data` type for `Model` is `any`. If we make our own custom type for `Model`, we can specify the structure for our `data` field

Reply:
I'm reluctant to do that as part of this PR, because balloons the scope. I think that belongs more as a task along with @alwayshooin 's data mapping.